third_party/nvfuser/csrc/transform_iter.h

```diff
@@ -39,80 +39,98 @@ struct id_int_lt {
-      //  this functionality could be useful.
-      bool replay_swizzle = false);
+      const std::vector<IterDomain*>& target_domain,
+      std::unordered_map<IterDomain*, IterDomain*> id_map);
+
+  ReplayTransformations& setErrorOnFailure(bool error_on_failure) {
+    error_on_failure_ = error_on_failure;
+    return *this;
+  }
+
+  ReplayTransformations& setReplaySwizzle(bool replay_swizzle) {
+    replay_swizzle_ = replay_swizzle;
+    return *this;
+  }
 
   // Replays outputs that were generated from ids.first on ids.second
   void runReplay();
 
   // Returns map from provided target domain to their corresponding IDs
   const std::unordered_map<IterDomain*, IterDomain*>& getReplay() {
-    if (!ran_replay)
+    if (!ran_replay_) {
       runReplay();
+    }
     return id_map_;
   }
 
   // Returns leaf_ids_ the size_t marks the order in which they were put into
   // the map, this is part of the structure because it's used to generate the
   // order from 'getLeafIDs'
   const std::unordered_map<IterDomain*, size_t>& getUnorderedLeafIDs() {
-    if (!ran_replay)
+    if (!ran_replay_) {
       runReplay();
+    }
     return leaf_ids_;
   }
 
   // Returns all terminating IDs that resulted from the replay. Leaf IDs are run
   // to run deterministic, but otherwise in no specific order.
   const std::vector<IterDomain*>& getLeafIDs() {
-    if (!ran_replay)
+    if (!ran_replay_) {
       runReplay();
+    }
     return leaf_vec_;
   }
+
+ protected:
+  using IterVisitor::handle;
+
+  // Transform dispatch
+  void handle(Expr* e) override;
+
+  // We're going to replay this split operation on the corresponding ID
+  void handle(Split* s) override;
+
+  // We're going to replay this merge operation on the corresponding IDs
+  void handle(Merge* m) override;
+
+  // We're going to replay this swizzle operation on the corresponding IDs
+  //  if replaying swizzle is enabled.
+  void handle(Swizzle2D* m) override;
+
+  size_t newCounter() {
+    return counter_++;
+  }
+
+ protected:
+  // NOLINTNEXTLINE(cppcoreguidelines-non-private-member-variables-in-classes)
+  const std::vector<IterDomain*>& target_domain_;
+  // NOLINTNEXTLINE(cppcoreguidelines-non-private-member-variables-in-classes)
+  std::unordered_map<IterDomain*, IterDomain*> id_map_;
+  // NOLINTNEXTLINE(cppcoreguidelines-non-private-member-variables-in-classes)
+  std::unordered_map<IterDomain*, size_t> leaf_ids_;
+
+ private:
+  bool error_on_failure_ = true;
+
+  // Indicates if we want to replay swizzle ops on the replayed
+  //  tensor.
+  // The swizzle op will be replayed if true,
+  // The swizzle inputs will be directly forwarded, and therefore skipping
+  //  the swizzle op if false.
+  // Currently this options should always be off but
+  //  later we may have cases in scheduling large fusions where
+  //  this functionality could be useful.
+  // NOLINTNEXTLINE(cppcoreguidelines-non-private-member-variables-in-classes)
```

Comment:
Is this still needed? It is private now.

Reply:
Deleted